cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,119 +11,149 @@
-                    }
+                String expr = exprBuilder.toString();
 
-                    arguments.get(0).visitAll(new NodeVisitorAdapter() {
-                        @Override
-                        public void accept(ExpressionNode node) {
-                            processExpression(node, suspiciousPoints);
-                        }
-                    });
-
-                    for (int i = 1; i < arguments.size(); i++) {
-                        arguments.get(i).visitAll(new NodeVisitorAdapter() {
-                            @Override
-                            public void accept(ExpressionNode node) {
-                                processExpression(node, suspiciousPoints);
-                            }
-
-                            @Override
-                            public void accept(ArgumentNode node) {
-                                // not to hardcore?
-                                String argument = node.getArgument();
-                                processUsage(argument, suspiciousPoints);
-                            }
-                        });
-                    }
+                Node substitution = substitutions.get(expr);
 
-                    return;
+                if (substitution != null) {
+                    substitution.visitAll(this);
+                } else {
+                    processUsage(expr);
+                    sb.append('*').append(expr);
                 }
+            }
 
-                node.visitAll(new NodeVisitorAdapter() {
-                    @Override
-                    public void accept(ExpressionNode node) {
-                        processExpression(node, suspiciousPoints);
-                    }
-
-                    @Override
-                    public void accept(ArgumentNode node) {
-                        // not to hardcore?
-                        String argument = node.getArgument();
-                        processUsage(argument, suspiciousPoints);
-                    }
-                });
+            @Override
+            public void accept(ConstantNode node) {
+                String value = node.getValue();
+                if (value.endsWith("\n")) {
+                    value = value.replace("\n", ""); // todo this hack should be replaced by fix in parser!
+                }
+                sb.append(value);
             }
         });
+
+        return sb.toString();
     }
 
-    private void processExpression(ExpressionNode node, Set<Node> suspiciousPoints) {
-        if (node.getKey() != null) {
-            return;
-        }
+    void processUsage(ArgumentNode argumentNode) {
+        getValue(argumentNode);
 
-        // fixme dirty
-        String expression = node.getExpression();
-        int expressionStart = expression.indexOf('{');
-        int expressionEnd = expression.lastIndexOf('}');
+        processUsage(getArgument(argumentNode));
+    }
+
+    protected void processUnknownUsage(String variable) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1186)